AppSyncRealTimeClient/Support/CountdownTimer.swift

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// A resettable timer that executes `onCountdownComplete` closure after
+/// `interval`.
+///
+/// The timer will execute the closure on a background queue. If the closure
+/// includes work that must be performed on a specific queue, make sure to dispatch
+/// it inside the closure.
+class CountdownTimer {
```

Comment:
This implementation looks good! It looks a lot cleaner since we no longer have to play around with timestamps every 5 minutes to determine stale connection. In the description i saw the previous problem was it did not get cleared before we go to the background. How does this implementation solve the problem with going to background?

Reply:
Previously, the consumer/caller was instructed to call `unsubscribe` during a background lifecycle event, which would eventually invoke `disconnect`. However, that did not stop the async work item from being processed. The new code path `cancel`s the work item upon unsubscribe, which prevents it from being invoked upon the app returning to foreground.

Reply:
Or at least, it does now that I fixed the bug in the previous version of the PR where I was calling `websocket.disconnect()` instead of `self.disconnect()` when there were no more listeners. :)